src/listing.rs

```diff
@@ -104,15 +127,21 @@ pub fn directory_listing<S>(
     req: &HttpRequest<S>,
     skip_symlinks: bool,
     random_route: Option<String>,
-    sort_method: SortingMethods,
-    reverse_sort: bool,
 ) -> Result<HttpResponse, io::Error> {
     let title = format!("Index of {}", req.path());
     let base = Path::new(req.path());
     let random_route = format!("/{}", random_route.unwrap_or_default());
     let is_root = base.parent().is_none() || req.path() == random_route;
     let page_parent = base.parent().map(|p| p.display().to_string());
 
+    let mut sort_method: Option<SortingMethod> = None;
+    let mut sort_order: Option<SortingOrder> = None;
+
+    if let Ok(query) = Query::<QueryParameters>::extract(req) {
```

Comment:
Could you do  this without the `mut` with `let (sort_method, sort_order) = if let ...`?

Reply:
And, in fact, couldn't you put this into the handler signature like in the *Query* example [here](https://actix.rs/docs/extractors/)?

Reply:
I'll try, it didn't seem that easy when I first tried. `files_listing_renderer` takes an `HttpRequest` as parameter, so I was quite stuck with that...